tests/notebooks/training-integration.ipynb

```diff
@@ -0,0 +1,904 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Name:\", pytorchjob.metadata.name)\n",
+    "print(\"Spec:\", pytorchjob.spec.pytorch_replica_specs)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### List existing PyTorchJob\n",
+    "\n",
+    "List PyTorchJob in the current namespace."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "[job.metadata.name for job in client.list_pytorchjobs()]"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create PyTorchJob\n",
+    "\n",
+    "Create a PyTorchJob using the SDK."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client.create_pytorchjob(pytorchjob)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### Get PyTorchJob\n",
+    "Get the created PyTorchJob by name and check its data.  \n",
+    "Make sure that it completes successfully before proceeding. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "raises-exception"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# verify that the Job was created successfully\n",
+    "# raises an error if it doesn't exist\n",
+    "pytorchjob = client.get_pytorchjob(name=PYTORCHJOB_NAME);"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "raises-exception"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# wait for the Job to complete successfully\n",
+    "assert_job_succeeded(client, PYTORCHJOB_NAME, job_kind=\"PyTorchJob\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "print(\"Job:\", pytorchjob.metadata.name, end=\"\\n\\n\")\n",
+    "print(\"Job Spec:\", pytorchjob.spec, sep=\"\\n\", end=\"\\n\\n\")\n",
+    "print(\"Job Status:\", pytorchjob.status, sep=\"\\n\", end=\"\\n\\n\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Get PyTorchJob Training logs\n",
+    "Get and print the training logs of the PyTorchJob with the training steps "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print_training_logs(client, PYTORCHJOB_NAME, container=\"pytorch\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Delete PyTorchJob\n",
+    "\n",
+    "Delete the created PyTorchJob and check that all created resources were removed as well."
```

Comment:
Again, we don't really check for other "created resources", so that part of the comment can go.